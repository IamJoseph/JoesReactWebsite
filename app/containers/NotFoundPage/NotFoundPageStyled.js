import styled from 'styled-components';
import planet from '../../assets/planet.png';

export const Container = styled.section`
  align-items: center;
  background: url(${planet}), hsla(51, 35%, 90%, .5);
  background-position: 20% 30%;
  background-repeat: no-repeat;
  background-size: cover;
  box-shadow: 0px 0px 7px 0px #a2a2a2;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  margin-left: 25px;
  margin-right: 25px;
  max-width: 1460px;
  overflow-y: auto;
  padding: 0 .5em;
  width: 100%;
  z-index: 1;
  @media(min-width: 768px){
    padding: 0 1em;
  }
  @media(min-width: 1150px){
    padding: 0 1.87em;
  }
`;

export const H1 = styled.h1`
  font-family: 'Coda', sans-serif;
`;
