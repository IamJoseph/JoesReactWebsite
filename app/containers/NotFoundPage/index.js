/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { Container, H1 } from './NotFoundPageStyled';

export default class NotFound extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <Container>
        <H1>OOPS you took a wrong turn!</H1>
        <h2>Sorry about that :(</h2>
      </Container>
    );
  }
}
