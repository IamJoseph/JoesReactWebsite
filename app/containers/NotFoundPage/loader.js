/**
 * Asynchronously loads the components for NotFoundPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/NotFoundPage')
    .then(cb)
    .catch(errorLoading);
};
