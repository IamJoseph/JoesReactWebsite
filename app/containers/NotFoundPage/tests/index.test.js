import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NotFoundPage from '../index';

describe('<NotFoundPage />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <NotFoundPage />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
