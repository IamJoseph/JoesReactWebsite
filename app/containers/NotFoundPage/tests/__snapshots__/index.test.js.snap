// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render correctly 1`] = `
<styled.section>
  <styled.h1>
    OOPS you took a wrong turn!
  </styled.h1>
  <h2>
    Sorry about that :(
  </h2>
</styled.section>
`;
